### ISSUE TITLE
feature: add `copilot job enable` and `disable` commands

### ISSUE THREAD

**Issue by bvtujo** [aws/copilot-cli#2429]
To minimize context switching, we should add a `copilot job disable` command to stop a job's schedule rule from executing. 

This way, we can 
```bash
$ copilot job disable
Which job would you like to disable? 
    db
  > report-generator
Job is only deployed in one environment; defaulting to *test*.
Job report-generator has been disabled and will not run again until you enable it.
```

We should also offer a `copilot job invoke` command to run jobs on demand, even disabled ones.

Related: #1823

**Comment by huanjani** [comment]
Let's keep this in mind when designing `job status`!

**Comment by kangere** [comment]
@bvtujo will you be accepting PRs for this issue?

**Comment by kangere** [comment]
@huanjani will you be accepting PRs for this issue?

**Comment by huanjani** [comment]
Hi, @kangere! Yes, that would be amazing!

Please follow the guide [here](https://github.com/aws/copilot-cli/blob/mainline/CONTRIBUTING.md) and feel free to comment in this Issue with your feature design; that way we can make your work more efficient by providing any guidance early on.

Thanks so much!

**Comment by kangere** [comment]
@huanjani  I was going to go with the proposed design above by @bvtujo 

> To minimize context switching, we should add a `copilot job disable` command to stop a job's schedule rule from executing.
> 
> This way, we can
> 
> ```shell
> $ copilot job disable
> Which job would you like to disable? 
>     db
>   > report-generator
> Job is only deployed in one environment; defaulting to *test*.
> Job report-generator has been disabled and will not run again until you enable it.
> ```
> 
> We should also offer a `copilot job invoke` command to run jobs on demand, even disabled ones.
> 
> Related: #1823

In addition to this, I propose have optional flags for job name and env e.g 

`copilot job disable --job-name report-generator --env test`

If the above flags are not passed in, the cli would default to the interactive mode that was proposed above by @bvtujo . The above examples would be the same for the `copilot job enable` command . Let me know what you think

**Comment by huanjani** [comment]
Looking back at #1823, I see you have been waiting patiently for this for a long time! 🙈 

One small piece of feedback: let's do ` -n, --name string         Name of the job.` as with the other commands. 
`--app` and `--help` flags would be great, too, but we can add those if you don't want to!

Will you also do `copilot job enable` at the same time? (`copilot job invoke` could be a separate thing.)

Thanks so much for taking this on!

**Comment by huanjani** [comment]
@kangere, what do you think about this?:
Rather than adding two commands, `job enable` and `job disable`, we could add `'none'` as a possible value for [on.schedule](https://aws.github.io/copilot-cli/docs/manifest/scheduled-job/#on-schedule), which would disable the event rule.

So the manifest file would become:
```yaml
on:
  schedule: 'none'
```
The reason we're opting to this route is because we want all infrastructure changes to flow through CloudFormation instead of using the SDK.

**Comment by kangere** [comment]
@huanjani  no problem, I did not have time to take this on till now so I don't mind the wait. 

I think the second approach to update the job in the manifest file makes more sense, It gives better visibility into the current status of the job without having to fire off any commands or check the aws console. It also has the advantage of being version controlled. 

I think we should go with the second approach

**Comment by kangere** [comment]
@huanjani in addition, I think `copilot job invoke` should still be implemented as a command since it acts more as a trigger for an action rather than a state of the infrastructure. My suggestion would be to  implement this as a separate issue

**Comment by kangere** [comment]
@huanjani could you point me to the files that would need to be changed? I can't seem to find the job related files

**Comment by huanjani** [comment]
Yes, of course!

Poke around in these files. I can send more files/guidance tomorrow if you'd like.

https://github.com/aws/copilot-cli/tree/mainline/internal/pkg/template/templates/workloads/jobs/scheduled-job has the CFN template and manifest template
https://github.com/aws/copilot-cli/blob/mainline/internal/pkg/template/templates/workloads/partials/cf/eventrule.yml
https://github.com/aws/copilot-cli/blob/mainline/internal/pkg/manifest/job.go
https://github.com/aws/copilot-cli/blob/mainline/internal/pkg/manifest/validate.go#L335
https://github.com/aws/copilot-cli/blob/mainline/internal/pkg/deploy/cloudformation/stack/scheduled_job.go

**Comment by kangere** [comment]
@huanjani after looking into this a bit, I propose adding a field to the manifest file called `enabled` for example:

```yaml
enabled: true
```
It would be true by default and could be set to false. 

The reason I suggest this is because the field `schedule` in the manifest is passed in as a parameter to cloudformation so setting it to `none` would be an invalid cloudformation input for the Eventbridge rule and the previous `schedule` value would not be retained since it would be overwritten by the new `none` value.  Let me know what you think

**Comment by huanjani** [comment]
Hi, @kangere!

Thanks for the suggestion! We'd like to avoid adding a top-level field, though, and 'none' in the `on.schedule` field makes logical sense to me.

You make a good point about retaining the previous `schedule` value, but perhaps users could just comment out that line when adding 
```
on: 
  #schedule: "@weekly"
  schedule: 'none'
```
What do you think about adding a new CloudFormation parameter in https://github.com/aws/copilot-cli/blob/mainline/internal/pkg/deploy/cloudformation/stack/scheduled_job.go called `Enabled` that has a boolean value that depends on whether or not `on.schedule == none` and is then used in https://github.com/aws/copilot-cli/blob/mainline/internal/pkg/template/templates/workloads/jobs/scheduled-job/cf.yml?

**Comment by kangere** [comment]
@huanjani  setting the `on.schedule` field to `none` makes sense and  adding the new cloudformation parameter  called `Enabled` also makes sense, I'm not opposed to either of these suggestions. The part I see an issue is when the `on.schedule` field is set to `none`, cloudformation is still expecting the `Schedule` parameter which in this case will be set to the value of `none`, which in turn is referenced by the Eventbridge Rule property called `ScheduleExpression`. Passing the value `none` here would lead to an error,  do you understand my concern or am I missing something ?

**Comment by efekarakus** [comment]
Apologies on the late response @kangere !

I think we can set to pretty much anything, maybe just `rate(5 minutes)` and have the `STATE: DISABLED`

**Comment by kangere** [comment]
@efekarakus That makes sense . Thanks for answering my concerns

**Comment by huanjani** [comment]
This feature is now released in v1.17.0: https://github.com/aws/copilot-cli/releases/tag/v1.17.0! 🎉 
Thank you so much, @kangere! ❤️
